### ISSUE TITLE
UWP Webview2 Size shrinks - Extended displays

### ISSUE THREAD

**Issue by contactteckee** [MicrosoftEdge/WebView2Feedback#3002]
Iam trying with a simple test UWP app which is having a webview2 control , noticed an issue where in the extended monitor , if i minimize the app and restore the app , the webview control shrinks in size. Please see the attachment. 
if i resize the main window then the control docks properly.
I tried using Grid instead of frame , still the issue exists. Any ideas to fix this issue. I guess this is something to do with DPI and not sure how it can be done within webview2.

Primary display is set with 1920X1080 resolution and scale of 125% and the Extended monitor is set with same resolution and scale of 100% (windows recommended).

Reference of the issue in Microsoft Q&A - 
https://learn.microsoft.com/en-us/answers/questions/1102111/uwp-webview2-size-shrinks-extended-displays.html

Framework: UWP
OS: Win10/Win11

![image](https://user-images.githubusercontent.com/119040388/204012856-bf0bcfc9-e5f6-4f72-9722-19ce11731105.png)

Below is the sample code.
```
xmlns:xamlcontrols="using:Microsoft.UI.Xaml.Controls"

 <Grid>
         <Grid.RowDefinitions>
               
             <RowDefinition Height="*"></RowDefinition>
         </Grid.RowDefinitions>
         <Grid.ColumnDefinitions>
             <ColumnDefinition Width="*"></ColumnDefinition>
                
         </Grid.ColumnDefinitions>
          
    
         <Frame  Grid.Row="0"  Grid.Column ="0" Margin="5 5 5 5" >
             <xamlcontrols:WebView2 x:Name="WebView2" Source="https://learn.microsoft.com/en-us/microsoft-edge/webview2/" />
         </Frame>
     </Grid>
```

[AB#43277742](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/43277742)

**Comment by bradp0721** [comment]
I've marked the bug as tracked so it's added to our internal bug priority list. Does this issue only happen for UWP? Does it repro with the [WebView2APISample](https://github.com/MicrosoftEdge/WebView2Samples/tree/main/SampleApps/WebView2APISample) app?

Since resizing the window fixes the WebView2 size, as a workaround, you could have your app code resize the WebView2 when it is restored. You likely will need to change the WebView2 size for it to have an effect. Have you tried the workaround provided in the Microsoft Q&A you linked above?

**Comment by contactteckee** [comment]
@bradp0721  Thank you for prioritizing this.

The temporary workaround provided in the Microsoft QA won’t help for UWP as it involves hardcoding the resolution parameters and it may vary from user to user

I tried with the Webview2 git samples in the extended display and the issue can be reproduced in certain scenarios.
1)	Winforms
When moved from the primary to extended monitor , the issue can be replaced. In the extended monitor , minimize and restore the drawing happens properly.
![image](https://user-images.githubusercontent.com/119040388/219289134-f3b6f3fc-ea0f-47b1-835e-04c30cf0c357.png)

2) WPF The webview works properly when moving to extended monitor , minimize/restore

![image](https://user-images.githubusercontent.com/119040388/219289434-88b6bfe3-fd7e-4003-803d-653b70c52d0f.png)

Hope these answers your questions.

**Comment by contactteckee** [comment]
@nishitha-burman ,@bradp0721 , how can i know the status of this issue?

**Comment by DotNet-Fan** [comment]
@champnic  I haven't used the canary runtimes and rely mainly on nuget package. Are there any standalone package/zip that can be downloaded instead of changing the development environment locally for canary builds?

**Comment by contactteckee** [comment]
@champnic  @bradp0721 

The issue still not fixed with the canary build. When i minimize and maximize from the extended monitor , the webview2 control shrinks.

Microsoft.UI.Xaml - 2.8.2 /2.8.4

Install-Package Microsoft.Web.WebView2 -Version 1.0.1774.30

Microsoft Edge canary build is installed and the environment variables are set like this.

WEBVIEW2_BROWSER_EXECUTABLE_FOLDER -> C:\Users\<user>\AppData\Local\Microsoft\Edge SxS\Application\115.0.1886.0

WEBVIEW2_RELEASE_CHANNEL_PREFERENCE -> 1

**Comment by champnic** [comment]
@DotNet-Fan Normally you can set a registry key, but I forget offhand whether that works for UWP apps:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/set-preview-channel
But the easiest for UWP apps is to set the environment variable in the app.

@contactteckee Can you confirm that the version that's actually getting picked up by the app is 1886? You can look in Task Manager or Process Explorer for the command line of the child msedgewebview2.exe processes.

I'm reopening this issue though, as we've had another report of this fix not addressing all scenarios.

**Comment by contactteckee** [comment]
@champnic 

Just to summarize , these are my observations where the app shrinks

1) Minimize /Maximize from the extended display.
2) Move the window from the primary display to extended
3) Keeping the window stay open or minimized, go to Windows sleep mode and then come back. The webview2 control shrinks.
4) if the URL is loaded previously before Windows sleep when comes back, it becomes blank.

**Comment by champnic** [comment]
We just checked in a new fix in runtime versions 117.0.1950.0+. Could you try that and see if it's resolved for you?

**Comment by champnic** [comment]
Thanks for confirming, and glad it's working now. 117 will ship in Beta around 8/21, and Stable around 9/14. Thanks!

**Comment by contactteckee** [comment]
Ok thanks @champnic . When you say Stable around 9/14 , how does the update works after 9/14? The users have to install windows patch or update Edge browser?

**Comment by champnic** [comment]
If the user already has the WebView2 Runtime installed it will automatically update to the latest 117 stable version around 9/14. Users on Win10 and Win11 should already have the runtime installed, so there is no work needed here from your end. For Beta, the user would need to install the Edge Beta channel version and the app would need to be set to load that version.

**Comment by contactteckee** [comment]
@champnic  This is great as it will automatically update to the stable version. Users won't use the beta and thanks for the details.

**Comment by contactteckee** [comment]
@champnic any updates on this? 9/14 was the last date mentioned , Is this published?

**Comment by champnic** [comment]
@contactteckee The first fix mentioned has gone to stable, although we found a potential second cause for this issue and just put in another fix last week. That will ship to stable around December 7th. Were you still getting reports of this issue?

**Comment by champnic** [comment]
Secondary fix is in runtimes version 120.0.2187.0+ if you wanted to test using prerelease (though I know you said you couldn't previously).

**Comment by contactteckee** [comment]
Hello @champnic , So far I haven't seen any issues with the stable build. Hope this wont affect with the new secondary fixes.

Thanks
